File: Code/apps/problems_directory/problems/two-sum/drivers/javascript/driver.js
// --- DRIVER CODE START ---
const fs = require('fs');

const DELIMITER = '$$$DELIMITER$$$';

try {
    if (!fs.existsSync('/app/input.txt')) throw new Error('Input file not found');
    const inputData = fs.readFileSync('/app/input.txt', 'utf-8');
    
    const testCases = inputData.split(DELIMITER).filter(tc => tc.trim() !== '');

    testCases.forEach((testCase) => {
        const lines = testCase.trim().split('\n');
        
        // Arg 0: Smart Parse (JSON or Space-Separated)
        let arg0;
        const raw0 = lines[0].trim();
        if (raw0.startsWith('[')) {
            arg0 = JSON.parse(raw0);
        } else {
            arg0 = raw0.split(/\s+/).map(Number);
        }
        const arg1 = Number(lines[1]);

        if (typeof twoSum !== 'function') {
            throw new Error("Function 'twoSum' not found.");
        }
        
        const result = twoSum(arg0, arg1);
        console.log(result);
        console.log(DELIMITER);
    });

} catch (err) {
    console.error("Driver Error:", err.message);
    process.exit(1);
}
// --- DRIVER CODE END ---

File: Code/apps/problems_directory/problems/two-sum/drivers/python/driver.py
import sys
import json

DELIMITER = "$$$DELIMITER$$$"

def main():
    try:
        with open("/app/input.txt", "r") as f:
            input_data = f.read()
            
        test_cases = input_data.split(DELIMITER)
        sol = Solution()
        
        for test_case in test_cases:
            if not test_case.strip(): continue
            lines = test_case.strip().split('\n')
            
            raw_val = lines[0].strip()
            if raw_val.startswith("["):
                arg0 = json.loads(raw_val)
            else:
                arg0 = [int(x) for x in raw_val.split()]
            arg1 = int(lines[1])

            result = sol.twoSum(arg0, arg1)
            
            # Sort lists to ensure [4,9] matches [9,4]
            if isinstance(result, list):
                result.sort()
            
            # Use json.dumps to print [1,2] without spaces
            print(json.dumps(result, separators=(',', ':')))
            print(DELIMITER)
            
    except Exception as e:
        sys.stderr.write(str(e))

if __name__ == "__main__":
    main()